### PR TITLE
Reuse S3FileSystem instance

### PR DESCRIPTION
**Pull request by majetideepak**
Improve the S3FileSystem support by reusing the S3FileSystem instance. Rename S3FileTest.cpp to S3FileSystemTest.cpp

The following changes are also made.

- Clarify that the S3FileSystem owns the S3Client instance. Update the S3ReadFile API accordingly.

- Add a fileHandle test as it is used by the HiveConnector

- Fix a benign bug and re-factor the S3FileSystemTest

### CODE REVIEW

File: velox/connectors/hive/storage_adapters/s3fs/tests/S3FileTest.cpp [hunk 1]
```diff
@@ -136,3 +137,24 @@ TEST_F(S3FileSystemTest, viaRegistry) {
   auto readFile = s3fs->openFileForRead(s3File);
   readData(readFile.get());
 }
+
+TEST_F(S3FileSystemTest, fileHandle) {
```
Comment (mbasmanova) [inline velox/connectors/hive/storage_adapters/s3fs/tests/S3FileTest.cpp:141]: I noticed a mis-match between the test name and file name: S3FileTest.cpp vs. S3FileSystemTest. Would you fix in a follow-up PR?

File: velox/connectors/hive/storage_adapters/s3fs/tests/S3FileTest.cpp [hunk 2]
```diff
@@ -136,3 +137,24 @@ TEST_F(S3FileSystemTest, viaRegistry) {
...
+  const char* bucket_name = "data3";
```
Comment (mbasmanova) [inline velox/connectors/hive/storage_adapters/s3fs/tests/S3FileTest.cpp:142]: naming: bucketName

File: velox/connectors/hive/storage_adapters/s3fs/tests/S3FileTest.cpp [hunk 3]
```diff
@@ -136,3 +137,24 @@ TEST_F(S3FileSystemTest, viaRegistry) {
...
+  const char* file = "test.txt";
+  const std::string filename = localPath(bucket_name) + "/" + file;
+  const std::string s3File = s3URI(bucket_name) + "/" + file;
+  filesystems::registerS3FileSystem();
```
Comment (mbasmanova) [inline velox/connectors/hive/storage_adapters/s3fs/tests/S3FileTest.cpp:146]: Should this go into setup method?

File: velox/connectors/hive/storage_adapters/s3fs/tests/S3FileTest.cpp [hunk 4]
```diff
@@ -136,3 +137,24 @@ TEST_F(S3FileSystemTest, viaRegistry) {
...
+  addBucket(bucket_name);
+  {
+    LocalWriteFile writeFile(filename);
+    writeData(&writeFile);
+  }
+  auto hiveConnectorConfigs = minioServer_->hiveConfig();
```
Comment (mbasmanova) [inline velox/connectors/hive/storage_adapters/s3fs/tests/S3FileTest.cpp:152]: this code repeats a lot; perhaps extract a helper method

File: velox/connectors/hive/storage_adapters/s3fs/tests/S3FileTest.cpp (hunk 1 above)
Comment (majetideepak) [reply velox/connectors/hive/storage_adapters/s3fs/tests/S3FileTest.cpp:135]: Fixed here.

File: velox/connectors/hive/storage_adapters/s3fs/tests/S3FileTest.cpp (hunk 3 above)
Comment (majetideepak) [reply velox/connectors/hive/storage_adapters/s3fs/tests/S3FileTest.cpp]: done.

File: velox/connectors/hive/storage_adapters/s3fs/tests/S3FileSystemTest.cpp [hunk 5]
```diff
@@ -87,52 +89,63 @@ class S3FileSystemTest : public testing::Test {
...
+  const char* bucketName = "data3";
+  const char* file = "test.txt";
+  const std::string filename = localPath(bucketName) + "/" + file;
+  const std::string s3File = s3URI(bucketName) + "/" + file;
+  addBucket(bucketName);
+  {
+    LocalWriteFile writeFile(filename);
+    writeData(&writeFile);
+  }
+  auto hiveConfig = minioServer_->hiveConfig();
```
Comment (mbasmanova) [inline velox/connectors/hive/storage_adapters/s3fs/tests/S3FileSystemTest.cpp:145]: hiveConfig doesn't seem to be used here. Am I missing something?

**Pull request closed without merging**